ga4gh/datarepo.py

```diff
@@ -0,0 +1,176 @@
+"""
+The backing data store for the GA4GH server
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import os
+
+import ga4gh.exceptions as exceptions
+import ga4gh.datamodel.datasets as datasets
+import ga4gh.datamodel.references as references
+
+
+class AbstractDataRepository(object):
+    """
+    An abstract GA4GH data repository
+    """
+    def __init__(self):
+        self._datasetIdMap = {}
+        self._datasetNameMap = {}
+        self._datasetIds = []
+        self._referenceSetIdMap = {}
+        self._referenceSetNameMap = {}
+        self._referenceSetIds = []
+
+    def addDataset(self, dataset):
+        """
+        Adds the specified dataset to this backend.
```

Comment:
The comments are out of date here and elsewhere. Probably %s/backend/data repository/g will do the trick